plutus-contract/src/Language/Plutus/Contract/Tx.hs

```diff
@@ -80,7 +83,25 @@ data UnbalancedTx = UnbalancedTx
         deriving stock (Eq, Show, Generic)
         deriving anyclass (Aeson.FromJSON, Aeson.ToJSON, IotsType)
 
-Lens.TH.makeLenses ''UnbalancedTx
+instance Semigroup UnbalancedTx where
+    tx1 <> tx2 = UnbalancedTx {
+        _inputs = _inputs tx1 <> _inputs tx2,
+        _outputs = _outputs tx1 <> _outputs tx2,
+        _forge = _forge tx1 <> _forge tx2,
+        _requiredSignatures = _requiredSignatures tx1 <> _requiredSignatures tx2,
+        _dataValues = _dataValues tx1 <> _dataValues tx2,
+        _validityRange = _validityRange tx1 /\ _validityRange tx2,
+        _valueMoved = _valueMoved tx1 <> _valueMoved tx2
+        }
+
+instance Plutus.Semigroup UnbalancedTx where
+    tx1 <> tx2 = tx1 <> tx2 -- using Prelude.Semigroup
```

Comment:
I don't particularly object to having more instances, but implementing them in terms of the Haskell ones seems bad. And we can't implement one without using the Haskell ones because we'd need one for `Set`.

Is this used a lot?

Reply:
I guess it's "never" going to be used in on-chain code... but it makes me feel uneasy.

Reply:
Good point about `Set`, I could change it to list but that's not the right thing either. Maybe I'll just leave it as it was.